src/portfft/utils.hpp

```diff
@@ -172,11 +180,15 @@ std::vector<sycl::kernel_id> get_transpose_kernel_ids() {
  */
 template <typename T>
 inline std::shared_ptr<T> make_shared(std::size_t size, sycl::queue& queue) {
-  return std::shared_ptr<T>(sycl::malloc_device<T>(size, queue), [captured_queue = queue](T* ptr) {
-    if (ptr != nullptr) {
-      sycl::free(ptr, captured_queue);
-    }
-  });
+  T* ptr = sycl::malloc_device<T>(size, queue);
+  if (ptr != nullptr) {
+    return std::shared_ptr<T>(ptr, [captured_queue = queue](T* ptr) {
+      if (ptr != nullptr) {
+        sycl::free(ptr, captured_queue);
+      }
+    });
+  }
+  throw internal_error("Could not allocate usm memory of size: ", size * sizeof(T), " bytes");
```

Comment:
```suggestion
  if (ptr == nullptr) {
    throw internal_error("Could not allocate usm memory of size: ", size * sizeof(T), " bytes");
  }
  return std::shared_ptr<T>(ptr, [captured_queue = queue](T* ptr) {
    if (ptr != nullptr) {
      sycl::free(ptr, captured_queue);
    }
  });
```
This would look better with less nesting.